packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -262,9 +376,56 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
           if (!balance) throw Fail`Result lacked balance key: ${result}`;
           return harden(toChainAmount(balance));
         },
+
+        withdrawRewards() {
+          throw assert.error('Not implemented');
+        },
+
+        /**
+         * @param {Delegation[]} delegations
+         */
+        async undelegate(delegations) {
+          trace('undelegate', delegations);
+          const { helper } = this.facets;
+          const { chainAddress, bondDenom, timer } = this.state;
+
+          const result = await E(helper.owned()).executeEncodedTx(
+            delegations.map(d =>
+              toAnyJSON(
+                MsgUndelegate.toProtoMsg({
+                  delegatorAddress: chainAddress.address,
+                  validatorAddress: d.validatorAddress,
+                  amount: { denom: bondDenom, amount: d.shares },
+                }),
+              ),
+            ),
+          );
+
+          const response = tryDecodeResponse(
+            result,
+            MsgUndelegateResponse.fromProtoMsg,
+          );
+          trace('undelegate response', response);
+          const { completionTime } = response;
+          const endTime = BigInt(completionTime.valueOf() / 1000);
+
+          const maxClockSkew = 10n * 60n;
```

Comment:
Maybe make this a constant at the top of the file and add jsdoc doc comment to `undelegate` that explains: 
1) this promise wont resolve until the unbonding period is over
2) a buffer of 10 minutes is added to completionTime to account for clock skew